index.d.ts

```diff
@@ -0,0 +1,41 @@
+declare module 'react-swipeable' {
+  import * as React from 'react'
+
+  namespace ReactSwipeable {
+    interface Event<T> {
+      event: React.TouchEvent<T>
+      deltaX: number
+      deltaY: number
+      absX: number
+      absY: number
+      velocity: number
+      dir: 'Left' | 'Right' | 'Up' | 'Down'
+    }
+
+    type Callback<T extends Element = HTMLElement> = ({}: Event<T>) => void
```

Comment:
from what I can tell i think the `<T>` seems to be a bit overkill? Appears to be "generics", honestly beyond me, https://www.typescriptlang.org/docs/handbook/generics.html.

But since we have control over input and output we should be able to define these explicitly?

Also do you know exactly what is happening with the `extends Element = HTMLElement`? The `EventData` returned from a `Callback` is explicit, and doesn't extend anything.

Reply:
Same as https://github.com/dogfessional/react-swipeable/pull/133#discussion_r269016818